diff --git a/src/main/scala/huancun/DataStorage.scala b/src/main/scala/huancun/DataStorage.scala
index 14d264e..1db7213 100644
--- a/src/main/scala/huancun/DataStorage.scala
+++ b/src/main/scala/huancun/DataStorage.scala
@@ -43,7 +43,7 @@ class DataStorage(implicit p: Parameters) extends HuanCunModule {
   /* Define some internal parameters */
   val nrStacks = 2
   val stackBits = log2Ceil(nrStacks)
-  val bankBytes = 8
+  val bankBytes = 1
   val rowBytes = nrStacks * beatBytes
   val nrRows = sizeBytes / rowBytes
   val nrBanks = rowBytes / bankBytes
diff --git a/src/main/scala/huancun/HCCacheParameters.scala b/src/main/scala/huancun/HCCacheParameters.scala
index 98026d1..0e95a7b 100644
--- a/src/main/scala/huancun/HCCacheParameters.scala
+++ b/src/main/scala/huancun/HCCacheParameters.scala
@@ -125,7 +125,7 @@ case class HCCacheParameters
 ) {
   require(ways > 0)
   require(sets > 0)
-  require(channelBytes.d.get >= 8)
+  // require(channelBytes.d.get >= 8)
   require(dirReadPorts == 1, "now we only use 1 read port")
   if (!inclusive) {
     require(clientCaches.nonEmpty, "Non-inclusive cache need to know client cache information")
diff --git a/src/main/scala/huancun/HuanCun.scala b/src/main/scala/huancun/HuanCun.scala
index 123c369..5f740d7 100644
--- a/src/main/scala/huancun/HuanCun.scala
+++ b/src/main/scala/huancun/HuanCun.scala
@@ -221,7 +221,7 @@ class HuanCun(implicit p: Parameters) extends LazyModule with HasHuanCunParamete
             fifoId = None
           )
         },
-        beatBytes = 32,
+        beatBytes = 1,
         minLatency = 2,
         responseFields = cacheParams.respField,
         requestKeys = cacheParams.reqKey,
@@ -361,7 +361,7 @@ class HuanCun(implicit p: Parameters) extends LazyModule with HasHuanCunParamete
 
     val slices = wrapper.node.in.zip(wrapper.node.out).zipWithIndex.map {
       case (((in, edgeIn), (out, edgeOut)), i) =>
-        require(in.params.dataBits == out.params.dataBits)
+        // require(in.params.dataBits == out.params.dataBits)
         val rst = if(cacheParams.level == 3 && !cacheParams.simulation) {
           ResetGen()
         } else reset
